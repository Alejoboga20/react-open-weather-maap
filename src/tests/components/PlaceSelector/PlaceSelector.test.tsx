import { shallow } from 'enzyme';
import {
  PlaceSelector,
  PlaceSelectorProps,
} from '../../../components/PlaceSelector/PlaceSelector';

const placeSelectorProps: PlaceSelectorProps = {
  inputButtons: [
    {
      label: 'Location Test',
      value: 'LT',
      location: { latitude: 10, longitude: 100 },
    },
  ],
  onClick: jest.fn(() => console.log('click')),
  blockButtons: false,
};

describe('Place Selector Tests', () => {
  const wrapper = shallow(<PlaceSelector {...placeSelectorProps} />);

  test('should Match Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should block buttons', () => {
    const wrapper = shallow(
      <PlaceSelector {...placeSelectorProps} blockButtons={true} />
    );

    const blockButton = wrapper.find('InputButton').prop('disabled');
    expect(blockButton).toBe(true);
  });

  test('should call onClick function', () => {
    wrapper.find('InputButton').simulate('click');
    expect(placeSelectorProps.onClick).toHaveBeenCalled();
  });
});
