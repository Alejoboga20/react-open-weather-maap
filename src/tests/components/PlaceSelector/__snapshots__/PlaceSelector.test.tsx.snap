// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Place Selector Tests should Match Snapshot 1`] = `
<div
  className="radio_container"
>
  <InputButton
    disabled={false}
    key="LT"
    label="Location Test"
    location={
      Object {
        "latitude": 10,
        "longitude": 100,
      }
    }
    onClick={[MockFunction]}
    value="LT"
  />
</div>
`;
